Intermittent failure in `spdm_crypt_lib_tests`

Related issue: https://github.com/DMTF/libspdm/issues/3393.

```
[==========] spdm_crypt_lib_tests: Running 1 test(s).
[ RUN      ] libspdm_test_fips
[  ERROR   ] --- status
[   LINE   ] --- /home/runner/work/libspdm/libspdm/unit_test/test_spdm_fips/test_spdm_fips.c:29: error: Failure!
FFDH compute key failed 
[  FAILED  ] libspdm_test_fips
[==========] spdm_crypt_lib_tests: 1 test(s) run.
[  PASSED  ] 0 test(s).
[  FAILED  ] spdm_crypt_lib_tests: 1 test(s), listed below:
[  FAILED  ] libspdm_test_fips
```
https://github.com/DMTF/libspdm/blob/fbb6a05a947899d1e050ce2a97630f2e296dc2e2/library/spdm_crypt_lib/fips/libspdm_selftest_ffdh.c#L79-L97

--- a/os_stub/cryptlib_openssl/pk/dh.c
+++ b/os_stub/cryptlib_openssl/pk/dh.c
@@ -1,6 +1,6 @@
 /**
  *  Copyright Notice:
- *  Copyright 2021-2022 DMTF. All rights reserved.
+ *  Copyright 2021-2026 DMTF. All rights reserved.
  *  License: BSD 3-Clause License. For full text see link: https://github.com/DMTF/libspdm/blob/main/LICENSE.md
  **/
 
@@ -579,7 +579,16 @@ bool libspdm_dh_compute_key(void *dh_context, const uint8_t *peer_public_key,
         goto cleanup;
     }
 
-    *key_size = secret_len;
+    if (secret_len > final_key_size) {
+        goto cleanup;
+    }
+
+    /* Pad with leading zeros.
+     * OpenSSL strip leading zero bytes from the raw DH result. */
+    memmove(key + final_key_size - secret_len, key, secret_len);
+    libspdm_zero_mem(key, final_key_size - secret_len);
+
+    *key_size = final_key_size;
     result = true;
 
 cleanup:

